logstash-core/src/main/java/org/logstash/instrument/witness/Witness.java

```diff
@@ -39,17 +43,28 @@ public Witness() {
         this.reloadWitness = new ReloadWitness();
         this.eventsWitness = new EventsWitness();
         this.pipelinesWitness = new PipelinesWitness();
+        this.processWitness = new ProcessWitness();
+        this.processWitnessScheduler = new WitnessScheduler(processWitness);
     }
 
     /**
      * This is a dirty hack since the {@link Witness} needs to mirror the Ruby agent's lifecycle which, at least for testing, can mean more then 1 instance per JVM, but only 1
      * active instance at any time.  Exposing this allows Ruby to create the instance for use in it's agent constructor, then set it here for all to use as a singleton.
      * <p>THIS IS ONLY TO BE USED BY THE RUBY AGENT</p>
      *
-     * @param __instance The instance of the {@link Witness} to use as the singleton instance that mirror's the agent's lifecycle.
+     * @param newInstance The instance of the {@link Witness} to use as the singleton instance that mirror's the agent's lifecycle.
      */
-    public static void setInstance(Witness __instance) {
-        _instance = __instance;
+    public static void setInstance(Witness newInstance) {
```

Comment:
@jakelandis might be a good idea to `synchronize` this method to avoid weird concurrency issues leaving dangling daemon threads around?

Reply:
Will do.  _FWIW, only the tests will ever call this more then once._